### PR TITLE
Validate regex when adding to the filter_token filter

### LINKED ISSUE #2093: Add regex validation when adding to filter_token bucket
In the past few days, there has been two instances of users adding a regex to the filter_token that wasn't valid, leading to #dev-alerts being spammed with regex-related errors.

This issue proposes that we should validate the regex when adding to the bucket, and if it's not valid, then we return an error message and don't add to the bucket to save these alerts.

In terms of implementation, should be a case of doing a try/except on a `re.compile`.

### PR DESCRIPTION
**Pull request by TizzySaurus**
Closes #2093.

When a user attempts to add a regex to the `filter_token` filter, the regex is now validated before being added to prevent errors when checking if a message matches the regex. If the regex isn't valid, an error message is sent informing the user that an invalid regex was passed, and the regex doesn't get added to the filter list.

### CODE REVIEW

**Review by mbaruh — COMMENTED** [review]
re.error is going to contain where it failed. Consider adding the error message to the output message. An example is: https://github.com/python-discord/bot/blob/7e8e95f07e343f1d7d9a8069b6cdb1a9fcbb00d7/bot/exts/moderation/clean.py#L56

**Comment by TizzySaurus** [comment]
> re.error is going to contain where it failed. Consider adding the error message to the output message. An example is:
> 
> https://github.com/python-discord/bot/blob/7e8e95f07e343f1d7d9a8069b6cdb1a9fcbb00d7/bot/exts/moderation/clean.py#L56

Yep! Added :+1:

File: bot/exts/filters/filter_lists.py [hunk 1]
```diff
@@ -72,6 +73,18 @@ async def _add_data(
         elif list_type == "FILE_FORMAT" and not content.startswith("."):
             content = f".{content}"
 
+        # If it's a filter token, validate the passed regex
+        elif list_type == "FILTER_TOKEN":
+            try:
+                _ = re.compile(content)
```
[the remark this answers is not in the document]
Comment (TizzySaurus) [reply bot/exts/filters/filter_lists.py:79]: Yep, for some reason I forgot you don't have to assign.

File: bot/exts/filters/filter_lists.py [hunk 2]
```diff
@@ -72,6 +73,18 @@ async def _add_data(
...
+            except re.error as e:
+                await ctx.message.add_reaction("❌")
+                await ctx.send(
+                    f"{ctx.author.mention} that's not a valid regex! "
+                    f"Regex error message: {e.msg}."
+                )
```
[the remark this answers is not in the document]
Comment (ChrisLovering) [reply bot/exts/filters/filter_lists.py:85]: Yea true, probably fine though, not sure if it's needed, but not worth changing if it already exists elsewhere.

**Review by ChrisLovering — APPROVED** [review]
my only outstanding comment isn't required for this to work, so approving anyway.

**Pull request merged**